protostar/configuration_file/configuration_file.py

```diff
@@ -86,6 +88,32 @@ def set_command_names_provider(
     ):
         pass
 
+    @classmethod
+    def create_appending_cairo_path_suggestion(
+        cls, logger, project_root_path, libs_path, package_name
+    ):
```

Comment:
[Design] `cls` is not used. This method shouldn't log anything but return a string. The method implementation should be located in concrete classes `ConfigurationFileV1` and `ConfigurationFileV2` unless the error message is generic.
 
```suggestion
    @abstractmethod
    def create_appending_cairo_path_suggestion(self) -> str:
```

Reply:
> `cls` is not used

`cls` was required so this method can be in this class as far as I remember. We can use the `@staticmethod` here.

> This method shouldn't log anything but return a string

OK

> The method implementation should be located in concrete classes `ConfigurationFileV1` and `ConfigurationFileV2` unless the error message is generic.

Sorry, but I don't understand. It is generic, it doesn't contain anything specific for the concrete implementations, therefore it introduces mutual code for every implementation and thus should not be distributed into different classes.
What would be the purpose of putting separate methods doing basically the same thing in separate derived classes?

Reply:
> Sorry, but I don't understand. It is generic, it doesn't contain anything specific for the concrete implementations...

"shared_command_configs" is specific to V1